src/dialects/sqlite/query-interface.js

```diff
@@ -23,11 +23,41 @@ class SQLiteQueryInterface extends QueryInterface {
     const fields = await this.describeTable(tableName, options);
     delete fields[attributeName];
 
-    const sql = this.queryGenerator.removeColumnQuery(tableName, fields);
-    const subQueries = sql.split(';').filter(q => q !== '');
+    const savepointName = 'remove_column_savepoint';
 
-    for (const subQuery of subQueries) {
-      await this.sequelize.query(`${subQuery};`, { raw: true, ...options });
+    // Disable foreign key constraints for the next sequence of operations
+    await this.sequelize.query('PRAGMA foreign_keys = OFF;', options);
+    // Use SAVEPOINT to allow for nested transactions
+    await this.sequelize.query(`SAVEPOINT ${savepointName};`, options);
+
+    try {
+      const sql = this.queryGenerator.removeColumnQuery(tableName, fields);
+      const subQueries = sql.split(';').filter(q => q !== '');
+
+      for (const subQuery of subQueries) {
+        await this.sequelize.query(`${subQuery};`, { raw: true, ...options });
```

Comment:
can these be executed in parallel?

Reply:
These should be executed sequentially since the order of execution matters.